javascript/javascript1/week4/voiceAssistant.js

```diff
@@ -0,0 +1,84 @@
+let name = "";
+let toDoList = [];
+
+function getReply(command){
+    // lowercase command and split it into commandWords array
+   const commandWords = command.toLowerCase().split(" ");
+   let response = null;
+   let toDoText = "";
+
+   if (name !== "" && commandWords[0] == "hello"){
+    console.log("you've already introduced yourself") 
+    return 
+   }
+
+   switch(commandWords[0]) {
+      case "hello":
+        name = commandWords.pop();
+        response = "nice to meet you " + name; break;
+
+      case "what":
+        switch(true) {
+            case commandWords.includes("name"):{
+              if (name === ""){
+                response = "sorry! you have not told us your name yet!"; break;
+              }
+              response = `your name is ${name} `; break;}
+            case commandWords.includes("todo?"):
+              response = toDoList; break;
+            case commandWords.includes("today?"):
+              response = new Date().toDateString(); break;
+            case commandWords[3] == "+":
+              response = parseInt(commandWords[2]) + parseInt(commandWords[4]); break;
+            case commandWords[3] == "*":
+              response = commandWords[2] * commandWords[4]; break;
+            case commandWords[3] == "/":
+              response = commandWords[2] / commandWords[4]; break;
+            case commandWords[3] == "-":
+              response = commandWords[2] - commandWords[4]; break;
+        }
+        break;
+
+      case "add":
+          
+          for (let i = 1; i < commandWords.length - 3; i++){
+            toDoText += commandWords[i] + " ";
+          }
```

Comment:
Smart!
If you want to level up these, you can look into "regex", but it can be pretty hard to understand.

Reply:
thanks. yeah regex is difficult. I should put more time later on it.